discovery/sync_manager.go

```diff
@@ -380,7 +386,7 @@ func (m *SyncManager) syncerHandler() {
 			}
 			m.syncersMu.Unlock()
 
-			s.Start()
+			s.Start(ctx)
```

Comment:
what if the context is already canceled here, should we catch it like how we catch `case <-m.quit:`?

Reply:
i dont think we need to check it absolutely everywhere - we just need to check it in cases like `m.quit` where we may hang.

Reply:
> i dont think we need to check it absolutely everywhere - we just need to check it in cases like m.quit where we may hang.

Not sure I follow...don't we check `m.quit` below at L539?

> Preemptively checking might not prevent another goroutine from closing m.quit immediately after we've checked whether it's closed.

So pls help me understand here...I thought we used `ctx` here because we want the caller to abort some operations, and in this case `syncerHandler` should not never be aborted when the context is canceled?

Reply:
> and in this case syncerHandler should not never be aborted when the context is canceled?

It will be aborted because if it's ctx is being cancelled, then quit will also be closed. we can listen on both - but it is redundant

Reply:
see [this](https://github.com/lightningnetwork/lnd/pull/9685/files#r2032907933) comment below for more details on this